Formula/dsda-doom.rb

```diff
@@ -0,0 +1,55 @@
+class DsdaDoom < Formula
+  desc "Fork of prboom+ with a focus on speedrunning"
+  homepage "https://github.com/kraflab/dsda-doom"
+  url "https://github.com/kraflab/dsda-doom/archive/refs/tags/v0.24.3.tar.gz"
+  sha256 "d4cfc82eea029068329d6b6a2dcbe0b316b31a60af12e6dc5ad3e1d2c359d913"
+  license "GPL-2.0-only"
+  head "https://github.com/kraflab/dsda-doom.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "fluid-synth"
+  depends_on "libvorbis"
+  depends_on "mad"
+  depends_on "pcre"
+  depends_on "portmidi"
+  depends_on "sdl2"
+  depends_on "sdl2_image"
+  depends_on "sdl2_mixer"
+  uses_from_macos "zlib"
+
+  def install
+    # Patch for Linux builds until kraflab/dsda-doom#122 is merged and added to a release
+    inreplace "prboom2/src/d_deh.c", "uint64_t", "uint_64_t"
+
+    system "cmake", "-S", "prboom2", "-B", "build",
+                    "-DBUILD_GL=ON",
+                    "-DWITH_DUMB=OFF",
+                    "-DWITH_IMAGE=ON",
+                    "-DWITH_FLUIDSYNTH=ON",
+                    "-DWITH_MAD=ON",
+                    "-DWITH_PCRE=ON",
+                    "-DWITH_PORTMIDI=ON",
+                    "-DWITH_VORBISFILE=ON",
+                    "-DWITH_ZLIB=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build", "--config", "Release"
+    system "cmake", "--install", "build", "--config", "Release"
+  end
+
+  def caveats
+    <<~EOS
+      Note that this formula only installs a Doom game engine, and no
+      actual levels. The original Doom levels are still under copyright,
+      you can find the shareware versions on:
+      https://www.doomworld.com/classicdoom/info/shareware.php
+
+      To avoid specifying full path of the WAD files, add their directory
+      to the "DOOMWADPATH" environment variable.
```

Comment:
This isn't specific to Homebrew anymore, so I don't think we want these `caveats`. We only add new `caveats` when they are specific to Homebrew. (Existing ones have been grandfathered in, but we'll probably remove them at some point.)